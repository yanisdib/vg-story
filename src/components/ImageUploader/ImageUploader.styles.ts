import styled from 'styled-components';

import { ImageAdd } from '@styled-icons/remix-line';


export const PreviewCanvas = styled.img`
    max-width: 100%;
    max-height: 100%;
    object-fit: contain;
    object-position: center;
    margin: auto;

`;

export const Uploader = styled.label`
    display: flex;
    flex-direction: column;
    justify-content: center;
    row-gap: 20px;
    align-items: center;
    width: 100%;
    min-height: 35vh;
    padding: 20px;
    font-size: 16px;
    color: var(--form-input-placeholder-color);
    border: var(--form-input-border);
    border-radius: var(--form-input-border-radius);
    cursor: pointer;
`;

export const FileInput = styled.input`
    display: none;
`;

export const AddImageIcon = styled(ImageAdd)`
    height: 50px;
    color: var(--form-input-border-color);
`;