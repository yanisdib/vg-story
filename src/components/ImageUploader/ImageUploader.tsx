import { ImageAttributes } from '../../interfaces';

import {
    AddImageIcon,
    FileInput,
    PreviewCanvas,
    Uploader
} from './ImageUploader.styles';


interface ImageUploaderProps {
    image?: ImageAttributes;
    onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}


function ImageUploader({ image, onChange }: ImageUploaderProps)
    : JSX.Element {

    return (
        <>
            {image && (
                <PreviewCanvas
                    alt={image.alt}
                    data-testid="preview-canvas"
                    src={image.src}
                    title={image.title}
                />
            )}
            <Uploader htmlFor="image-uploader">
                <AddImageIcon />
                Choose an image for your story
                <small>
                    You can only upload images of .jpg, .jpeg and.png formats
                </small>
                <FileInput
                    type="file"
                    id="image-uploader"
                    onChange={onChange}
                    accept=".jpg, .jpeg, .png"
                />
            </Uploader>
        </>
    );
}


export default ImageUploader;