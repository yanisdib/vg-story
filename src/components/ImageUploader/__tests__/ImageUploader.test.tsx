import { render } from '@testing-library/react';

import ImageUploader from '../ImageUploader';


describe('ImageUploader component', () => {

    const onImageChange = jest.fn();
    const currentImage = { src: 'hello.jpg' };

    it('should render without crashing', () => {

        const { container } = render(<ImageUploader image={currentImage} onChange={onImageChange} />);

        expect(container).toMatchSnapshot();
    });

    it('should render no image if currentImage is empty', () => {

        const { queryAllByTestId } = render(<ImageUploader onChange={onImageChange} />);

        expect(queryAllByTestId('preview-canvas')).toHaveLength(0);
    });

    it('should render the current image inside preview canvas', () => {

        const { getByTestId } = render(<ImageUploader image={currentImage} onChange={onImageChange} />);

        expect(getByTestId('preview-canvas')).toBeInTheDocument();
        expect(getByTestId('preview-canvas')).toHaveAttribute('src', 'hello.jpg');
    });
});