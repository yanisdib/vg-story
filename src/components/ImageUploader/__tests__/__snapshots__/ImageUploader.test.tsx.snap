// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageUploader component should render without crashing 1`] = `
<div>
  <img
    class="sc-bczRLJ ekoizA"
    data-testid="preview-canvas"
    src="hello.jpg"
  />
  <label
    class="sc-gsnTZi kspNoQ"
    for="image-uploader"
  >
    <svg
      aria-hidden="true"
      class="StyledIconBase-ea9ulj-0 lbJwfL sc-hKMtZM dAfHWB"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M21 15v3h3v2h-3v3h-2v-3h-3v-2h3v-3h2zm.008-12c.548 0 .992.445.992.993V13h-2V5H4v13.999L14 9l3 3v2.829l-3-3L6.827 19H14v2H2.992A.993.993 0 012 20.007V3.993A1 1 0 012.992 3h18.016zM8 7a2 2 0 110 4 2 2 0 010-4z"
      />
    </svg>
    Choose an image for your story
    <small>
      You can only upload images of .jpg, .jpeg and.png formats
    </small>
    <input
      accept=".jpg, .jpeg, .png"
      class="sc-dkzDqf hVKfRa"
      id="image-uploader"
      type="file"
    />
  </label>
</div>
`;
